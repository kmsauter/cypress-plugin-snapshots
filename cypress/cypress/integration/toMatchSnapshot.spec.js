const { runSuites } = require('../fixtures/test-data');

describe('toMatchSnapshot', () => {
  it('html escaping', () => {
    cy.visit('/static/input.html')
      .then(() => {
        cy.get('#input-element').toMatchSnapshot();
      });
  });

  runSuites('toMatchSnapshot', function (item) {
    cy.wrap(item).toMatchSnapshot();
  });
});
