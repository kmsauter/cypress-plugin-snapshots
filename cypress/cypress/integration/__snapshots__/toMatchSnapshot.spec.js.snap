exports[`toMatchSnapshot > html escaping #0`] = `
<span class="input input--number input--required" id="input-element">
  <input
    type="number"
    name="new_address.house_number"
    value=""
    pattern="\\d*"
    required=""
    data-date-format="mm/dd/yyyy"
  />
</span>
`;

exports[`toMatchSnapshot > toMatchSnapshot > json > true #0`] =
true;

exports[`toMatchSnapshot > toMatchSnapshot > json > number #0`] =
5;

exports[`toMatchSnapshot > toMatchSnapshot > json > array #0`] =
[
  1,
  "string",
  false,
  null,
  "2019-02-11T05:00:00.000Z",
  {
    "a": 1,
    "b": 2
  },
  null,
  11
];

exports[`toMatchSnapshot > toMatchSnapshot > json > object #0`] =
{
  "array": [
    "string",
    5,
    true,
    null,
    {
      "a": 1,
      "b": 2
    },
    null
  ],
  "date": "2019-02-11T05:00:00.000Z",
  "false": false,
  "nested": {
    "array": [
      2,
      "string",
      true,
      null,
      null,
      "2019-02-11T05:00:00.000Z",
      {
        "a": 1,
        "b": 2
      }
    ],
    "number": 1,
    "true": true
  },
  "null": null,
  "string": "string"
};

exports[`toMatchSnapshot > toMatchSnapshot > html > jquery-get #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > jquery-create #0`] = `
<div>Created DIV</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > cy-get-element #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > DOM-exists #0`] = `
<div id="test-div">Test Div</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > DOM-created #0`] = `
<div>DOM-created</div>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > HTMLCollection #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > NodeList #0`] = `
<h4>H1</h4>
<h4>H2</h4>
<h4>H3</h4>
`;

exports[`toMatchSnapshot > toMatchSnapshot > html > svg #0`] = `
<svg height="20" width="40"><path d="M10 0 L0 20 L20 20 Z"></path></svg>
`;

exports[`toMatchSnapshot > toMatchSnapshot > json > false #0`] =
false;

exports[`toMatchSnapshot > toMatchSnapshot > json > null #0`] =
null;

exports[`toMatchSnapshot > toMatchSnapshot > json > string #0`] = `
kylie
`;

exports[`toMatchSnapshot > toMatchSnapshot > json > date #0`] = `
2019-02-11T05:00:00.000Z
`;
